samcli/commands/init/interactive_init_flow.py

```diff
@@ -40,29 +45,90 @@ def do_interactive(
     It will ask customers a few questions to init a template.
     """
     if app_template:
-        location_opt_choice = "1"
+        location_opt_choice = "2"
     else:
         click.echo("Which template source would you like to use?")
-        click.echo("\t1 - AWS Quick Start Templates\n\t2 - Custom Template Location")
-        location_opt_choice = click.prompt("Choice", type=click.Choice(["1", "2"]), show_choices=False)
-    if location_opt_choice == "2":
+        click.echo("\t1 - Hello World application\n\t2 - AWS Quick Start Templates\n\t3 - Custom Template Location")
+        location_opt_choice = click.prompt("Choice", type=click.Choice(["1", "2", "3"]), show_choices=False)
+
+        get_start_options(
+            location,
+            pt_explicit,
+            package_type,
+            runtime,
+            base_image,
+            dependency_manager,
+            output_dir,
+            name,
+            app_template,
+            no_input,
+            location_opt_choice,
+        )
+
+
+def get_start_options(
+    location,
+    pt_explicit,
+    package_type,
+    runtime,
+    base_image,
+    dependency_manager,
+    output_dir,
+    name,
+    app_template,
+    no_input,
+    location_opt_choice,
+):
+    if location_opt_choice == "1":
```

Comment:
I'm not a big fan of using if, else for control structure. It promotes more if else, if there are new options introduced.

Could we do something like dictionary dispatch? https://alysivji.github.io/quick-hit-dictionary-dispatch.html

Reply:
I think we're good here, not a big fan of this type of dictionary dispatch, we also have few conditions and I don't think it will grow to become unreadable/difficult to follow.

Reply:
After a discussion with @mgrandis, I'm okay with this as long as its a few choices. Can we document that we would need a refactor if the number of options increase, I'm concerned about possible bugs being introduced in the control flow with more options and using if else for managing them.

Reply:
I am not a big fan of if/else statements either, i believe it can easily explode to make more conditions, but thought it fitting for the case since they were fewer condition and it helped with readability in this case.